geode-core/src/main/java/org/apache/geode/internal/DistributionLocator.java

```diff
@@ -136,28 +139,27 @@ public static void main(String args[]) {
       }
 
       if (!Boolean.getBoolean(InternalDistributedSystem.DISABLE_SHUTDOWN_HOOK_PROPERTY)) {
-        final InetAddress faddress = address;
+        final HostAddress faddress = hostAddress;
         Runtime.getRuntime()
             .addShutdownHook(new LoggingThread("LocatorShutdownThread", false, () -> {
               try {
-                DistributionLocator.shutdown(port, faddress);
+                DistributionLocator.shutdown(port, faddress.getAddress());
               } catch (IOException e) {
                 e.printStackTrace();
               }
             }));
       }
 
-      lockFile = ManagerInfo.setLocatorStarting(directory, port, address);
+      lockFile = ManagerInfo.setLocatorStarting(directory, port,
+          hostAddress == null ? null : hostAddress.getAddress());
```

Comment:
Looks like this could be simplified by creating the `address` variable a bit earlier
```
final InetAddress address = hostAddress == null ? null : hostAddress.getAddress();
lockFile = ManagerInfo.setLocatorStarting(directory, port, address);
```